snapshots/snapshots-processor/src/main/kotlin/com/emergetools/snapshots/processor/utils/PreviewFinderUtils.kt

```diff
@@ -36,9 +36,44 @@ fun List<KSAnnotated>.functionsWithMultiPreviewAnnotation(
         .map { function ->
           function to getUniqueSnapshotConfigsFromMultiPreviewAnnotation(
             annotations = multiPreviewAnnotationPreviewAnnotations,
-            previewFunction = function
+            previewFunction = function,
           )
         }
     }
-    .toMap()
+
+  // The same function declaration can show up multiple times, so ensure the values are merged together
+  val mergedConfigs = mutableMapOf<KSFunctionDeclaration, List<ComposePreviewSnapshotConfig>>()
+  uniqueSnapshotConfigs.forEach {
+    mergedConfigs.putOrAppend(it.first, it.second)
+  }
+
+  return mergedConfigs
+}
+
+fun Resolver.getMultiPreviewAnnotations(): List<KSClassDeclaration> {
+  // Find all symbols with annotations and map to the annotation class declarations
+  val annotationClassDecls = getAllFiles()
```

Comment:
This could potentially introduce a perf regression unless you have a better idea on how to find all annotations within the module.

Reply:
Yeah I think this is the best bet, the only alternative would be to use `getSymbolsWithAnnotation`, but that requires knowing the FQN of the annotation at call time, which we don't have the luxury of having. 

Since `getAllFiles` just runs on the files within the module, I think this is fine as it limits the blast radius. From there you're filtering down to those we deem are annotation class declarations, then finding those in submodules, which IMO should not have a drastic perf implication since we're quite targeted at that point.

Reply:
Yea agreed, sounds good.